### PR TITLE
Format code & check formatting on CI

### LINKED ISSUE #31: Enable rustfmt checking locally and on Travis-ci

### PR DESCRIPTION
**Pull request by YaLTeR**
The new stable Rust is here along with rustfmt 1.3.0, which means it's finally time to ~~break everyone's PRs~~ format the code base and start keeping it that way.

Closes #31

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -125,97 +126,78 @@ static av1_tx_ind: [[usize; TX_TYPES]; TX_SETS] = [
   [1, 2, 3, 6, 4, 5, 7, 8, 9, 0, 0, 0, 0, 0, 0, 0],
   [3, 4, 5, 8, 6, 7, 9, 10, 11, 0, 1, 2, 0, 0, 0, 0],
   [7, 8, 9, 12, 10, 11, 13, 14, 15, 0, 1, 2, 3, 4, 5, 6],
-  [7, 8, 9, 12, 10, 11, 13, 14, 15, 0, 1, 2, 3, 4, 5, 6]
+  [7, 8, 9, 12, 10, 11, 13, 14, 15, 0, 1, 2, 3, 4, 5, 6],
 ];
 
 pub static max_txsize_rect_lookup: [TxSize; BlockSize::BLOCK_SIZES_ALL] = [
-      // 4X4
```
Comment (tdaede) [inline src/context.rs]: rustfmt skip this block

File: src/frame/plane.rs [hunk 2]
```diff
@@ -644,17 +633,14 @@ pub mod test {
   #[test]
   fn copy_from_raw_u8() {
     let mut plane = Plane::wrap(
-    vec![
```
Comment (tdaede) [inline src/frame/plane.rs]: rustfmt skip

File: src/frame/plane.rs [hunk 3]
```diff
@@ -695,15 +676,10 @@ pub mod test {
     plane.pad(4, 4);
     assert_eq!(
       &[
-        1u8, 1, 1, 2, 3, 4, 4, 4,
```
Comment (tdaede) [inline src/frame/plane.rs]: rustfmt skip

File: src/lrf.rs [hunk 4]
```diff
@@ -789,96 +941,118 @@ pub fn sgrproj_solve<T: Pixel>(set: u8, fi: &FrameInvariants<T>,
...
   let bit_depth = fi.sequence.bit_depth;
-  let round_h = if bit_depth == 12 {5} else {3};
-  let round_v = if bit_depth == 12 {9} else {11};
+  let round_h = if bit_depth == 12 { 5 } else { 3 };
+  let round_v = if bit_depth == 12 { 9 } else { 11 };
   let offset = 1 << (bit_depth + WIENER_BITS - round_h - 1);
   let limit = (1 << (bit_depth + 1 + WIENER_BITS - round_h)) - 1;
 
-  let mut work: [i32; SB_SIZE+7] = [0; SB_SIZE+7];
-  let vfilter: [i32; 7] = [ coeffs[0][0] as i32,
```
Comment (tdaede) [inline src/lrf.rs]: rustfmt skip

Comment (tdaede) [reply src/lrf.rs]: remove as i32?

File: src/recon_intra.rs [hunk 5]
```diff
@@ -28,101 +28,106 @@ static has_null: &'static [u8] = &[];
 //       . . o .
 //       . . . .
 static has_tr_4x4: &'static [u8] = &[
-  255, 255, 255, 255, 85, 85, 85, 85, 119, 119, 119, 119, 85, 85, 85, 85,
```
Comment (tdaede) [inline src/recon_intra.rs]: rustfmt skip

File: src/recon_intra.rs [hunk 6]
```diff
@@ -28,101 +28,106 @@ static has_null: &'static [u8] = &[];
...
-  15, 0, 5, 0, 7, 0, 5, 0,
-];
-static has_tr_16x64: &'static [u8] = &[ 255, 127 ];
-static has_tr_64x16: &'static [u8] = &[ 3, 1 ];
+static has_tr_8x32: &'static [u8] = &[255, 255, 127, 127, 255, 127, 127, 127];
+static has_tr_32x8: &'static [u8] = &[15, 0, 5, 0, 7, 0, 5, 0];
+static has_tr_16x64: &'static [u8] = &[255, 127];
+static has_tr_64x16: &'static [u8] = &[3, 1];
 
 static has_tr_tables: &'static [&'static [u8]] = &[
```
Comment (tdaede) [inline src/recon_intra.rs]: adjust comments

File: src/recon_intra.rs [hunk 7]
```diff
@@ -229,93 +247,96 @@ pub fn has_top_right(/*const AV1_COMMON *cm,*/
...
-  238, 78, 238, 14, 238, 78, 238, 14,
-];
-static has_bl_16x64: &'static [u8] = &[ 0, 0 ];
-static has_bl_64x16: &'static [u8] = &[ 42, 42 ];
+static has_bl_8x32: &'static [u8] = &[0, 1, 0, 0, 0, 1, 0, 0];
+static has_bl_32x8: &'static [u8] = &[238, 78, 238, 14, 238, 78, 238, 14];
+static has_bl_16x64: &'static [u8] = &[0, 0];
+static has_bl_64x16: &'static [u8] = &[42, 42];
 
 static has_bl_tables: &'static [&'static [u8]] = &[
```
Comment (tdaede) [inline src/recon_intra.rs]: adjust comments

**Comment by YaLTeR** [comment]
Fixed & rebased

**Comment by YaLTeR** [comment]
Removed the clippy commit because it's no longer necessary (that place fits in one line so it doesn't complain)

**Comment by YaLTeR** [comment]
[AWCY](https://beta.arewecompressedyet.com/?job=master-b2e0d946ede3360c9d40a14849555ce5979a9c6e&job=rustfmt-2019-08-16_153131-470d7a1)

**Pull request merged**